internal/handler/handler_test.go

```diff
@@ -230,3 +236,401 @@ func TestNewHtmlPdfService(t *testing.T) {
 		})
 	}
 }
+func TestUpload(t *testing.T) {
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	tests := []struct {
+		name         string
+		requestBody  string
+		setupFunc    func() (*http.Request, *htmlPdfService)
+		validateFunc func(*httptest.ResponseRecorder)
+	}{
+		{
+			name: "Success:: Upload",
+			setupFunc: func() (*http.Request, *htmlPdfService) {
+				b := new(bytes.Buffer)
+				y := multipart.NewWriter(b)
+				part, _ := y.CreateFormFile("file", "some-file")
+				part.Write([]byte("abc"))
+				y.Close()
+				r := httptest.NewRequest(http.MethodPost, "/v1/register", b)
+				r.Header.Set("Content-Type", y.FormDataContentType())
+				mockLogicier := mock.NewMockHtmlPdfServiceLogicIer(mockCtrl)
+				mockLogicier.EXPECT().Upload(gomock.Any()).Times(1).
+					DoAndReturn(func(f io.Reader) *respModel.Response {
+						gotData, err := ioutil.ReadAll(f)
+						if err != nil {
+							t.Error(err)
+							t.FailNow()
+						}
+						diff := testutil.Diff(gotData, []byte("abc"))
+						if diff != "" {
+							t.Error(testutil.Callers(), diff)
+						}
+						return &respModel.Response{
+							Status:  http.StatusCreated,
+							Message: "SUCCESS",
+							Data: map[string]interface{}{
+								"id": "1",
+							},
+						}
+					})
+				rec := &htmlPdfService{
+					logic: mockLogicier,
+				}
+				return r, rec
+			},
+			validateFunc: func(x *httptest.ResponseRecorder) {
+				if x.Code != http.StatusCreated {
+					t.Errorf("want %v got %v", http.StatusCreated, x)
+				}
+				var r respModel.Response
+				got, err := io.ReadAll(x.Body)
+				json.Unmarshal(got, &r)
```

Comment:
Unmarshal returns error. Handle the error for all test cases.

Reply:
done

Reply:
For error, current `return` needs to be replaced with `t.Errorf()`.

Reply:
done